UnicornRender/video/source/Primitives.cpp

```diff
@@ -132,5 +136,159 @@ Mesh& Primitives::Sphere(Mesh& mesh, float radius, uint32_t rings, uint32_t sect
 
     return mesh;
 }
+
+namespace
+{
+
+/**
+ * @brief Reads mesh from assimp and creates unicorn::Mesh.
+ * @param [in] mesh pointer to assimp mesh
+ * @param [in] scene assimp hierarhy scene
+ * @param [in] dir directory, where mesh is locating
+ *
+ * @return created unicorn::Mesh
+ */
+Mesh* ProcessMesh(aiMesh const* mesh, aiScene const* scene, std::string const& dir);
+
+/**
+* @brief Reads each node recursively and calls ProcessMesh for each aiMesh
+* @param [in] node node in scene hierarchy
+* @param [in] scene assimp hierarhy scene
+* @param [in] dir directory, where mesh is locating
+* @param [out] meshes fills list with create meshes
+*/
+void ProcessNode(aiNode const* node, aiScene const* scene, std::string const& dir, std::list<Mesh*>& meshes)
+{
+    assert(nullptr != node);
+    assert(nullptr != scene);
+
+    for (unsigned int i = 0; i < node->mNumMeshes; i++)
+    {
+        aiMesh* mesh = scene->mMeshes[node->mMeshes[i]];
+        meshes.push_back(ProcessMesh(mesh, scene, dir));
+    }
+    for (unsigned int i = 0; i < node->mNumChildren; i++)
+    {
+        ProcessNode(node->mChildren[i], scene, dir, meshes);
```

Comment:
It's most likely not going to be optimized away as a tail recursion. So at some point, you can easily catch a stack overflow here. Rewrite it as iteration would be a better choice.